src/drawing.cpp

```diff
@@ -156,10 +158,13 @@ int CalcDelayUntilNextFrame(double frameStartTime, uint16_t localPixelsDrawn, ui
 
     if (localPixelsDrawn > 0)
     {
-        const double minimumFrameTime = 1.0 / g_ptrSystem->EffectManager().GetCurrentEffect().DesiredFramesPerSecond();
-        double elapsed = g_Values.AppTime.CurrentTime() - frameStartTime;
-        if (elapsed < minimumFrameTime)
-            g_Values.FreeDrawTime = std::clamp(minimumFrameTime - elapsed, 0.0, 1.0);
+        const double fpsRaw = static_cast<double>(g_ptrSystem->EffectManager().GetCurrentEffect().DesiredFramesPerSecond());
+        // If FPS is invalid (<= 0 or non-finite), treat as unlimited (0s minimum frame time).
+        const double minimumFrameTime = (!std::isfinite(fpsRaw) || fpsRaw <= 0.0) ? 0.0 : (1.0 / fpsRaw);
```

Comment:
I don't think there is any way on our platform where a `size_t` (i.e. unsigned int) value could overflow the numerical range of `double`?